python/ctsm/crop_calendars/cropcal_module.py

```diff
@@ -49,11 +49,12 @@ def check_and_trim_years(year_1, year_n, ds_in):
     return ds_in
 
 
-def open_lu_ds(filename, year_1, year_n, existing_ds, ungrid=True):
+def open_lu_ds(filename, year_1, year_n, existing_ds, *, logger, ungrid=True):
     """
     Open land-use dataset
     """
     # Open and trim to years of interest
+    utils.log(logger, f"open_lu_ds(): Opening this_ds_gridded: {filename}")
```

Comment:
It took me a minute to figure this out but the "open_lu_ds(): " in the logger prefix is the name of the function. This seems useful to have. With the downside that the name needs to be maintained in case the subroutine name changes.

I'm not sure it's worth it, but in case the subroutine names change, you can use things like inspect.current_frame() or sys._getframe() to get the subroutine name. But, an idea to think about...